scripts/update_tags_per_languages.py

```diff
@@ -0,0 +1,365 @@
+    "categories": "category",
+    # "countries": "country",
+    # "labels": "label",
+    # "origins": "origin", 
+}
+
+dev = True
+if dev:
+    env = "net"
+    user = "off:off@" 
+else:
+    env = "org"
+    user = "" 
+
+headers = {
+    'Accept': 'application/json', 
+    'User-Agent': 'UpdateTagsLanguages',
+}
+data = {
+    'user_id': '',
+    'password': '',
+}
+
+logs_file = "update_tags_per_languages_logs"
+if os.path.isfile(logs_file):
+    os.remove(logs_file)
+
+mapping_languages_countries = {
+    "aa": "dj",
+    "ar": "world",  # ar but categories are en:<french name> 
+    "be": "by",
+    "bg": "bg",
+    "br": "fr",
+    "bs": "ba",
+    "ca": "fr",
+    "cs": "cz",
+    "da": "dk",
+    "de": "de",
+    "el": "gr",
+    "en": "world",
+    "xx": "world", # xx but categories are en:<french name>
+    "es": "es",
+    "et": "ee",
+    "fa": "ir",
+    "fi": "fi",
+    "fr": "fr",
+    "hr": "hr",
+    "id": "id",
+    "is": "is",
+    "it": "it",
+    "ja": "jp",
+    "lt": "lt",
+    "ms": "my",
+    "nb": "no",
+    "nl": "nl",
+    "pl": "pl",
+    "pt": "pt",
+    "ro": "ro",
+    "ru": "ru",
+    "sk": "sk",
+    "sl": "si",
+    "sr": "rs",
+    "sv": "se",
+    "th": "th",
+    "zh": "cn",
+}
+
+def get_from_api(get_call_url: str) -> dict:
+    """Send a GET request to the given URL.
+    
+    :param get_call_url: the URL to send the request to
+
+    :return: the API response
+    stop the code if returned status_code is different than 200
+    """
+    get_call_url_res = requests.get(
+        get_call_url,
+        headers=headers,
+    )
+
+    if get_call_url_res.status_code != 200:
+        print(f"ERROR: when calling api. {get_call_url_res.status_code} status code. url: {get_call_url}")
+        sys.exit()
+    
+    return get_call_url_res.json()
+
+
+def unknown_tags_taxonomy_comparison(all_tags: dict, file: str, tag_type: str) -> dict:
+    """Iterate over all referenced tags, filter unknowns tags, and search in taxonomy
+
+    :param all_tags: all_tags as returned by API call
+    :param file: path of the corresponding taxonomy file
+    :param tag_type: the current tag type (category, allergens, etc.). Used to save output file
+
+    :return: possible_wrong_language_tags {"old_tag": "tag_found_in_different_lang"}
+    """
+    possible_new_tags = []
+    possible_wrong_language_tags = {}
+    already_referenced_tags = []
+
+    for tag in all_tags["tags"]:
+        if tag["known"] == 0:
+            # limit number of iterations
+            if dev and len(possible_wrong_language_tags) > 0:
+                break
+
+            tag_name = tag['name']
+            # should retrieve all "en:blablabla, tag_name" or "it:tag_name"
+            tag_regex = re.compile(f'\n([a-z][a-z]:(?:[\w\s\-\']*\,-)*{tag_name})[,|\n]')
+
+            with open(file, "r") as f:
+
+                tag_regex_res = tag_regex.findall(f.read().lower().replace(" ", "-", -1))
```

Comment:
Performace: at this line you load all the file in memory, which is ok. But why not load it once and for all outside the for loop ? you could even use mmap for that: https://docs.python.org/3/library/mmap.html

Reply:
Let's stick to simple open() and not include mmap(), since we don't want to apply changes to the file content, but just read and search in it.